Ginger/GingerCoreCommon/VariablesLib/VariableSelectionList.cs

```diff
@@ -120,6 +185,58 @@ private ObservableList<OptionalValue> ConvertOptionalValuesStringToList(string v
             }
         }
 
+        public override List<string> GetExtraParamsDescription() 
+        {
+            List<string> extraParamsDescription = new List<string>();
+            extraParamsDescription.Add("Index=1");
+            extraParamsDescription.Add("GetLength=True");
+            return extraParamsDescription;
+        }
+
+
+        public override string GetValueWithParam(Dictionary<string, string> extraParamDict)
+        {
+            string param = string.Empty;
+            int index;
+            if (extraParamDict.Count == 0)
+            {
+                return Value;
+            }
+            foreach (KeyValuePair<string, string> keyValuePair in extraParamDict)
+            {
+                switch (keyValuePair.Key)
+                {
+                    case "Index":
+                        {
+                            bool isNumber = Int32.TryParse(keyValuePair.Value, out index);
+                            if (isNumber && index > 0 && OptionalValuesList.Count >= index)
+                            {
+                                return OptionalValuesList[index - 1].Value;
+                            }
+                            else
+                            {
+                                Reporter.ToUser(eUserMsgKey.ListOutOfBounds, "Index is out of bounds");
```

Comment:
no need to show message to user as pop up instead:
1- write the error to the Log with the variable name and wrong index
2- return value as the error, like: "Error!! variable index is out of bounds"